src/modules/ModalInterface.js

```diff
@@ -0,0 +1,158 @@
+     <textarea class="modal-textarea mb-2" placeholder="Your comment" cols="30" rows="10"></textarea>
+     <button type="submit" class="submit-button button mb-2">Comment</button>
+   </form>
+   </div>
+  </div>`;
+    modalBackground.innerHTML += '<span class="spacer p-2">a</span>';
+    mainElement.appendChild(modalBackground);
+  }
+
+  static updateCommentsCounter() {
+    const commentsContainer = document.getElementById('comments-container');
+    const allComments = commentsContainer.children;
+    const numberOfComments = commentsContainer.children.length;
+    const commentTitle = document.getElementById('modal-comments-title');
+
+    if (numberOfComments === 1 && (allComments[0].children[2].innerText || allComments[0].children[2].innerHTML) === 'No comments Yet') {
+      commentTitle.innerText = `Comments(${0})`;
+    } else {
+      commentTitle.innerText = `Comments(${numberOfComments})`;
+    }
+  }
+
+  static async displayComments(arrayOfComments, showIndex) {
+    if (!arrayOfComments) {
+      arrayOfComments = await Modal.getComments(showIndex);
+    }
+
+    const commentsContainer = document.getElementById('comments-container');
+    commentsContainer.innerHTML = '';
+    if (arrayOfComments) {
+      arrayOfComments.forEach((comment) => {
+        if (comment.username) {
+          commentsContainer.innerHTML += `<div class="comment-item d-flex">
+          <p class="me-1">${comment.creation_date}</p> 
+          <p class="me-1">${comment.username}:</p> 
+          <p class="me-1">${comment.comment}</p>
+          </div>`;
+        } else {
+          commentsContainer.innerHTML += `<div class="comment-item d-flex">
+          <p class="me-1">${comment.creation_date}</p> 
+          <p class="me-1">${comment.username}</p> 
+          <p class="me-1">${comment.comment}</p>
+          </div>`;
+        }
+      });
+    }
+
+    Modal.updateCommentsCounter();
+  }
+
+  static async postComment(id, name, newComment) {
+    const response = await fetch('https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/06PVGgvDd8phHuVhPNDL/comments', {
+      method: 'POST',
+      body: JSON.stringify({
+        item_id: id,
+        username: name,
+        comment: newComment,
+      }),
+      headers: {
+        'Content-type': 'application/json',
+
+      },
+    });
+
+    return response;
+  }
+
+  static async addNewComment(showIndex) {
+    const addNewCommentForm = document.querySelector('.comments-form-container');
+    if (addNewCommentForm) {
+      addNewCommentForm.addEventListener('submit', (e) => {
+        const nameInput = addNewCommentForm.children[0].value;
+        const commentTextArea = addNewCommentForm.children[1].value;
+        Modal.postComment(showIndex,
+          nameInput, commentTextArea)
+          .then(() => {
+            Modal.displayComments(null, showIndex);
+          });
+        e.preventDefault();
+      });
+    }
+  }
```

Comment:
**Changes Required**
- [ ] Fair effort on using the `async await`  promise syntactic sugar. However, you're mixing two syntax here, and the `async` keyword in this context is redundant. May we please remove the `.then` and stick to the `async await` format.

Reply:
In this one, I'm not really sure why I did this, but I think think that the await keyword doesn't work inside an addEventListener function, however I created a function that will handle that properly outside of this eventListener.